src/Components/CreateImageWizard/steps/Users/component/UserInfo.tsx

```diff
@@ -0,0 +1,199 @@
+  const [isPasswordVisible, setIsPasswordVisible] = useState(false);
+  const [isConfirmPasswordVisible, setIsConfirmPasswordVisible] =
+    useState(false);
+  const confirmUserPassword = useAppSelector(selectConfirmUserPassword);
+  const userSshKey = useAppSelector(selectUserSshKey);
+  const userAdministrator = useAppSelector(selectUserAdministrator);
+  const handleFieldChange = <K extends keyof UserWithAdditionalInfo>(
+    field: K,
+    value: UserWithAdditionalInfo[K],
+    setterAction?: ActionCreatorWithPayload<
+      string | boolean | undefined | number
+    >,
+    additionalLogic?: (updatedUser: UserWithAdditionalInfo) => void
+  ) => {
+    // Create updated user object
+    const updatedUser: UserWithAdditionalInfo = {
+      index,
+      name: userName,
+      password: userPassword,
+      confirmPassword: confirmUserPassword,
+      ssh_key: userSshKey,
+      administrator: userAdministrator,
+      [field]: value,
+    };
+    // Call specific setter action if provided
+    if (setterAction) {
+      dispatch(setterAction(value));
+    }
+    // Optional additional logic
+    if (additionalLogic) {
+      additionalLogic(updatedUser);
+    }
+    // Determine if user exists by index, not name
+    const userExists = users.some((user) => user.index === index);
+    if (!userExists) {
+      dispatch(addUser(updatedUser));
+    } else {
+      dispatch(editUser(updatedUser));
+    }
+  };
+  const togglePasswordVisibility = () => {
+    setIsPasswordVisible(!isPasswordVisible);
+  };
+  const toggleConfirmPasswordVisibility = () =>
+    setIsConfirmPasswordVisible((prevState) => !prevState);
+
+  const stepValidation = useUserValidation();
+  return (
+    <Form>
+      <FormGroup isRequired label="Username" fieldId="blueprint-user-name">
+        <HookValidatedInput
+          ariaLabel="blueprint user name"
+          value={userName}
+          onChange={(event, userName) =>
+            handleFieldChange('name', userName, setUserName)
+          }
+          placeholder="Enter username"
+          stepValidation={stepValidation}
+          fieldName="userName"
+        />
+      </FormGroup>
+      <FormGroup isRequired label="Password" fieldId="blueprint-user-password">
+        <HookValidatedInputWithButton
+          ariaLabel="blueprint user password"
+          value={userPassword || ''}
+          type={isPasswordVisible ? 'text' : 'password'}
+          onChange={(event, userPassword) =>
+            handleFieldChange('password', userPassword, setUserPassword)
+          }
+          placeholder="Enter password"
+          stepValidation={stepValidation}
+          fieldName="userPassword"
+          togglePasswordVisibility={togglePasswordVisibility}
+          isPasswordVisible={isPasswordVisible}
+          isEmpty={userPassword === ''}
+        />
+        <FormHelperText>
+          <HelperText>
+            <HelperTextItem>
+              Can contain at least 3 different character classes (uppercase
```

Comment:
I guess we require three different character classes, I'd change the wording a bit.

```suggestion
              Must contain at least 3 different character classes (uppercase
```